resources/resincli.robot

```diff
@@ -151,6 +155,15 @@ Check if "${option}" variable "${variable_name}" with value "${variable_value}"
     ${result} =  Run Process    echo "${result_env.stdout}" | grep ${variable_name} | grep " ${variable_value}"    shell=yes
     Process ${result}
 
+Check if "${option}" variable "${variable_name}" with value "${variable_value}" exists on "${type}" "${id}"
+    [Documentation]    Available values for argument ${option} are: ENV, CONFIG
+    @{list} =  Create List    ENV    CONFIG
+    Should Contain    ${list}    ${option}
+    ${result} =  Run Keyword If    '${option}' == 'ENV'    Run Process    balena envs --${type} ${id} --json | jq 'any(.[]; .name == "${variable_name}" and .value == "${variable_value}")' | grep true    shell=yes
+    ...    ELSE
+    ...    Run Process    balena envs --config --${type} ${id} --json | jq 'any(.[]; .name == "${variable_name}" and .value == "${variable_value}")' | grep true    shell=yes
```

Comment:
How should we test this? Do we know if jq is available in the runtime environment?

Reply:
It's really clunky to test - needs a PR in the e2e repo (e.g. https://github.com/balena-io/e2e/pull/581) to manually set the version. JQ is not in https://github.com/balena-io/autohat/blob/master/Dockerfile currently. That's where it needs to go.

Reply:
.. or you can just merge, it will get blocked if it doesn't work on the e2e repo after renovate bumps.

.. or just patch your BoB with the changes and see if it works there.

Reply:
the e2e test failed: https://balena-tests-s3-logs-bucket-1f454wud1z2ll.s3.amazonaws.com/8070885780--2845-1-11885e6dad2f4be28fffc3f62baba5/logs/log.html

Reply:
probably escaping: https://stackoverflow.com/questions/43904146/run-process-throws-error-expected-at-least-1-non-keyword-argument-got-0/44677600#44677600

Reply:
I've made some changes in a new commit, hopefully to fix the escaping. Will re-running the [e2e job](https://github.com/balena-io/e2e/actions/runs/8070885780/job/22049642958?pr=581) pick this up?

Reply:
yeah, that PR is pointing to the `build` tag, so as long as the build here is finished, re-running https://github.com/balena-io/e2e/pull/581 will test it (resin.io failure is expected).